lib/runs.js

```diff
@@ -64,4 +68,78 @@ async function* getIterator(options) {
   }
 }
 
-module.exports = { get, getAll, getIterator };
+
+// Fetches aligned runs from the wpt.fyi server, between the |from| and |to|
+// dates. If |experimental| is true fetch experimental runs, else stable runs.
+// Returns a map of date to list of runs for that date (one per product)
+//
+// TODO: Known problem: there are periods of time, mostly mid-late 2018, where
+// we ran both Safari 11.1 and 12.1, and the results are massively different.
+// We should fetch multiple runs for each browser and have upgrade logic.
+async function fetchAlignedRunsFromServer(products, from, to, experimental) {
+  const label = experimental ? 'experimental' : 'stable';
+  let params = `label=master&label=${label}`;
+  for (const product of products) {
+    params += `&product=${product}`;
+  }
+  const runsUri = `${RUNS_API}?aligned=true&max-count=1&${params}`;
+
+  console.log(`Fetching aligned runs from ${from.format('YYYY-MM-DD')} ` +
+      `to ${to.format('YYYY-MM-DD')}`);
+
+  let cachedCount = 0;
+  const before = moment();
+  const noCacheAfter = moment().subtract('3', 'days');
+  const alignedRuns = new Map();
+  while (from < to) {
+    const formattedFrom = from.format('YYYY-MM-DD');
+    from.add(1, 'days');
+    const formattedTo = from.format('YYYY-MM-DD');
+
+    // We advance the date (if necessary) before doing anything more, so that
+    // code later in the loop body can just 'continue' without checking.
+    from = advanceDateToSkipBadDataIfNecessary(from, experimental);
+
+    // Attempt to read the runs from the cache.
+    // TODO: Consider https://github.com/tidoust/fetch-filecache-for-crawling
+    let runs;
+    const cacheFile = path.join(path.join(__dirname, '..'),
+        `cache/${label}-${products.join('-')}-runs-${formattedFrom}.json`);
+    try {
+      runs = JSON.parse(await fs.promises.readFile(cacheFile));
+      if (runs.length) {
+        cachedCount++;
+      }
+    } catch (e) {
+      // No cache hit; load from the server instead.
+      const url = `${runsUri}&from=${formattedFrom}&to=${formattedTo}`;
+      const response = await fetch(url);
```

Comment:
(tangential comment... have you tested this w/ the latest node-fetch version? there are no tests to verify that 
https://github.com/web-platform-tests/results-analysis/pull/123 is safe, but the usage here seems so simple)

Reply:
(no, I haven't)